### ISSUE TITLE
False require-v-for-key warning for simple HTML markup

### ISSUE THREAD

**Issue by tankist** [vuejs/eslint-plugin-vue#449]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **eslint-plugin-vue Version:** 4.4.0
* **Node Version:** 8.9.3

**Please show your full configuration:**

```
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint',
    ecmaVersion: 2017
  },
  env: {
    browser: true,
    es6: true
  },
  // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
  // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
  extends: ['plugin:vue/recommended', 'plugin:vue/essential'],
  // required to lint *.vue files
  plugins: [
    'import',
    'vue',
    'prettier',
  ],
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: {
    // don't require .vue extension when importing
    'import/extensions': ['error', 'always', {
      js: 'never',
      vue: 'never'
    }],
    // disallow reassignment of function parameters
    // disallow parameter object manipulation except for specific exclusions
    'no-param-reassign': ['error', {
      props: true,
      ignorePropertyModificationsFor: [
        'state', // for vuex state
        'acc', // for reduce accumulators
        'e' // for e.returnvalue
      ]
    }],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': ['error', {
      optionalDependencies: ['test/unit/index.js']
    }],
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'vue/html-indent': ['error', 4],
    'vue/name-property-casing': ['error', 'kebab-case'],
    'vue/html-self-closing': ['error', {
      'html': {
        'void': 'any',
        'normal': 'any',
        'component': 'always'
      },
      'svg': 'always',
      'math': 'always'
    }]
  }
};
```

**What did you do? Please include the actual source code causing the issue.**

I've tried to render `<dl><dt><dd>` list with v-for in it

```vue

<template>
    <dl>
        <template v-for="prop in someArrayOfObjects">
            <dt>{{ prop.title }}</dt>
            <dd>{{ prop.value }}</dd>
        </template>
    </dl>
</template>
```

**What did you expect to happen?**

Valid `<dl><dt><dd> list`

**What actually happened? Please include the actual, raw output from ESLint.**

```
 WARNING  Compiled with 1 warnings                                                                                                                                                                                                      17:52:47


  ✘  https://google.com/#q=vue%2Frequire-v-for-key  Elements in iteration expect to have 'v-bind:key' directives
  src\pages\room.vue:6:17
                  <dt>{{ prop.title }}</dt>
                   ^

  ✘  https://google.com/#q=vue%2Frequire-v-for-key  Elements in iteration expect to have 'v-bind:key' directives
  src\pages\room.vue:7:17
                  <dd>{{ prop.value }}</dd>
                   ^


✘ 2 problems (2 errors, 0 warnings)


Errors:
  2  https://google.com/#q=vue%2Frequire-v-for-key

You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.    
```

Actually, Vue does not require :key for simple HTML, only for components: https://vuejs.org/v2/guide/list.html#key:

```
It is recommended to provide a key with v-for whenever possible, unless the iterated DOM content is simple, or you are intentionally relying on the default behavior for performance gains.
```

**Comment by Lukino2000** [comment]
Hi, I made the same mistake, see #447 and [this link](https://jsfiddle.net/b4oyragL/)
